wfl/select/stratify.py

```diff
@@ -0,0 +1,91 @@
+import warnings
+
+import numpy as np
+import random
+
+
+def test_short_distance(structure, mindist = 0.1):
+    """
+    Function for examining if given structure contains too short distance between atoms. 
+    mindGist is set to 0.3 Angs which safely avoids any error from FHI-AIMS. 
+
+    Parameters : 
+    ------------
+    structure : ase Atoms object
+        Structure to be examined. 
+    mindist : float
+        Minimum allowed distance between atoms. 
+
+    Return : 
+    Boolean
+
+    """
+    distancematrix = structure.get_all_distances()
+    distancematrix[np.diag_indices(distancematrix.shape[0])] = 1
+    if np.min(distancematrix) < mindist:
+        return False
+    else:
+        return True
+
+
+def stratified_random(inputs, outputs, num=5):
```

Comment:
That's a much more specific thing than just "stratifying" implies.  For this to be more generally useful I think it would need to be generalized.

One possibility is to write a general selection method that picks some number of configs from each category (the usual meaning of stratify is to subselect keeping the same relative population in each category, which isn't what you want here), with the category coming from an arbitrary `Atoms.info` (could default to `"config_type"`), and something like a dict with the keys being the category values and the values being the number of configs to select (or a fraction, with the total # in a separate argument?).

Then your top level script could make sure that the info field contained the category as you define it (global vs. local vs. random MD), and passes the dict and/or total number to get the 5 configs you want.

Reply:
Note that I haven't looked at all as to whether any of the existing selection methods do any of this sort of thing.

Reply:
I heard that `by_bool_func` in `select/simple.py` can actually accommodate this kind of sampling. I wasn't aware of it. So I'll just remove `stratify.py` from this PR and this sampling could be rather added to top-level code which is more specific to the individual project.

Reply:
I don't really see how `by_bool_func` can do this sampling, because I thought it just makes a yes/no decision about each config independent of all the others, but I'm happy to leave it up to you how you do the sampling you need for your top level code. I just want anything that gets incorporated into wfl itself to be at least somewhat general.